### PR TITLE
IDs are now restfull for account requests.

### LINKED ISSUE #220: Accepting a request doesn't work
It appears as if the Web client isn't properly submitting the group_id

### PR DESCRIPTION
**Pull request by MoritzHahn1337**
Fixes #220
Makes ids in the account requests restful
Also fixes group change problems

### CODE REVIEW

**Comment by MoritzHahn1337** [comment]
I think this also fixes #218

File: datameta/views/admin.py [hunk 1]
```diff
@@ -50,11 +50,11 @@ def v_admin_put_request(request):
     # Parse request
     accept = False
     try:
-        reg_req_id             = int(request.json_body['id'])
+        reg_req_id             = str(request.json_body['id'])
```
Comment (lkuchenb) [inline datameta/views/admin.py:49]: `json_body` is a string dict
```suggestion
        reg_req_id             = request.json_body['id']
```

File: datameta/views/admin.py [hunk 2]
```diff
@@ -50,11 +50,11 @@ def v_admin_put_request(request):
...
         response               = request.json_body['response']
         newuser_make_admin     = bool(request.json_body.get("group_admin"))
         newuser_fullname       = request.json_body['fullname'] # Admin can edit upon confirm
-        newuser_group_id       = int(request.json_body['group_id']) if request.json_body['group_id'] is not None else None # Admin can edit upon confirm
+        newuser_group_id       = str(request.json_body['group_id']) if request.json_body['group_id'] is not None else None # Admin can edit upon confirm
```
Comment (lkuchenb) [inline datameta/views/admin.py:53]:
```suggestion
        newuser_group_id       = request.json_body['group_id'] if request.json_body['group_id'] is not None else None # Admin can edit upon confirm
```

File: datameta/views/admin.py [hunk 3]
```diff
@@ -67,17 +63,17 @@ def v_admin_put_request(request):
         raise HTTPBadRequest()
 
     # Check if the request exists
-    reg_req = db.query(RegRequest).filter(RegRequest.id==reg_req_id).one_or_none()
+    reg_req = db.query(RegRequest).filter(RegRequest.uuid==reg_req_id).one_or_none()
```
Comment (lkuchenb) [inline datameta/views/admin.py:66]: Could use the `resource` module if we wanted to be more flexible here

File: datameta/views/admin.py [hunk 4]
```diff
@@ -67,17 +63,17 @@ def v_admin_put_request(request):
...
 
     # Check if the requesting user is authorized. Both the request group as well as the
     # administratively selected group have to be the requesting user's group
-    if not req_user.site_admin and (reg_req.group_id!=req_user.group_id or newuser_group_id!=req_user.group_id):
+    if not req_user.site_admin and (reg_req.group_id!=req_user.group_id.uuid or newuser_group_id!=req_user.group_id.uuid):
         return HTTPUnauthorized()
 
     # Check if the specified group id is valid
-    db_group = db.query(Group).filter(Group.id==newuser_group_id).one_or_none() if newuser_group_id is not None else None
+    db_group = db.query(Group).filter(Group.uuid==newuser_group_id).one_or_none() if newuser_group_id is not None else None
```
Comment (lkuchenb) [inline datameta/views/admin.py:76]: Same here

File: datameta/views/admin.py [hunk 5]
```diff
@@ -154,14 +150,13 @@ def v_admin_get(request):
     # If the requesting user is only a group admin, return only those users
     # that are in the same group
     if not req_user.site_admin:
-        query = query.filter(User.group_id==req_user.group_id)
+        query = query.filter(User.group.uuid==req_user.group_id)
```
Comment (lkuchenb) [inline datameta/views/admin.py:153]: `req_user` is a database object
```suggestion
        query = query.filter(User.group.uuid==req_user.group_id)
```
```suggestion
        query = query.filter(User.group_id==req_user.group_id)
```

File: datameta/views/admin.py [hunk 6]
```diff
@@ -154,14 +150,13 @@ def v_admin_get(request):
...
 
     response = {}
 
     response["users"] = [ {
-        'uuid' : str(user.uuid),
-        'id' : user.site_id,
-        'group_id' : user.group.site_id,
+        'id' : {'uuid' : str(user.uuid), 'site_id' : user.site_id},
+        'group_id' : {'uuid' : str(user.group.uuid), 'site_id' : user.group.site_id},
```
Comment (lkuchenb) [inline datameta/views/admin.py:159]: The entity schema is `{'uuid' : [...], 'site' : [...]}`
```suggestion
        'id' : {'uuid' : str(user.uuid), 'site' : user.site_id},
        'group_id' : {'uuid' : str(user.group.uuid), 'site' : user.group.site_id},
```

File: datameta/views/admin.py [hunk 7]
```diff
@@ -173,25 +168,25 @@ def v_admin_get(request):
     # If the requesting user is a site admin, return all groups, otherwise only theirs
     if req_user.site_admin:
         response['groups'] = [ { 
-            'uuid': str(group.uuid),
-            'site_id' : group.site_id,
+            'id' : {'uuid': str(group.uuid), 'site_id' : group.site_id },
```
Comment (lkuchenb) [inline datameta/views/admin.py:171]:
```suggestion
            'id' : {'uuid': str(group.uuid), 'site' : group.site_id },
```
```suggestion
            'id' : {'uuid': str(group.uuid), 'site' : group.site_id },
```

File: datameta/views/admin.py [hunk 8]
```diff
@@ -173,25 +168,25 @@ def v_admin_get(request):
...
             'name': group.name
             } for group in db.query(Group) ]
     else:
         response['groups'] = [ { 
-            'uuid': str(group.uuid),
-            'site_id' : group.site_id,
+            'id' : {'uuid': str(group.uuid), 'site_id' : group.site_id },
```
Comment (lkuchenb) [inline datameta/views/admin.py:176]:
```suggestion
            'id' : {'uuid': str(group.uuid), 'site' : group.site_id },
```
```suggestion
            'id' : {'uuid': str(group.uuid), 'site' : group.site_id },
```

File: datameta/views/admin.py [hunk 9]
```diff
@@ -173,25 +168,25 @@ def v_admin_get(request):
...
     query = db.query(RegRequest)
     if not req_user.site_admin:
         query = query.filter(RegRequest.group_id==req_user.group_id)
 
-    response['reg_requests'] = [ { key : getattr(row, key) for key in
-        [ 'id','fullname', 'email', 'group_id', 'new_group_name' ] } for row in query ]
-
-
+    response['reg_requests'] = [ { 
+        'id' : str(reg_request.uuid),
```
Comment (lkuchenb) [inline datameta/views/admin.py:185]: Not using the schema here?

**Review by lkuchenb — CHANGES_REQUESTED** [review]
Thanks!

File: datameta/static/js/admin.js [hunk 10]
```diff
@@ -149,7 +149,7 @@ DataMeta.admin.subnav = function() {
     // Handle registration request preselection
     var showreq = DataMeta.uilocal.showreq;
     DataMeta.uilocal.showreq = null;
-    if (showreq != null) {
+    if (!(showreq == '')){
```
Comment (lkuchenb) [inline datameta/static/js/admin.js:152]: Not a fan of `!=`? ^^

**Review by lkuchenb — APPROVED** [review]

**Pull request merged**